addons/sale_loyalty/models/sale_order.py

```diff
@@ -356,7 +358,7 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
                 tax_str=len(discountable_per_tax) and any(t.name for t in mapped_taxes[tax]) and _(' - On product with the following taxes: %(taxes)s', taxes=", ".join(mapped_taxes[tax].mapped('name'))) or '',
             ),
             'product_id': reward.discount_line_product_id.id,
-            'price_unit': -(price * discount_factor),
+            'price_unit': -min((price * discount_factor), max_discount_allowed),
```

Comment:
I find it strange that you integrate the `max_discount_allowed` here, in the middle of a logic split by taxes, which conflicts with the fact that `max_discount_allowed` is considered according to the whole SO logic.

Can't you consider it directly where `discount_factor` is set/computed ?

Reply:
I could, but I don't find it clearer honestly. :thinking: I'll try to see if I change my mind though.
Note that I didn't know that it was already considered elsewhere, so I'll investigate a bit more in that direction too.